src/components/rating/rating.tsx

```diff
@@ -53,6 +52,20 @@ export class Rating
   /** The component's value. */
   @Prop({ reflect: true, mutable: true }) value = 0;
 
+  @Watch("value")
+  setValue(newValue: number, oldValue: number): void {
+    if (newValue == oldValue && !this.required) {
```

Comment:
I don't think `newValue == oldValue` will ever be true since the `@Watch` only fires on changes to the newValue. So this if statement probably will never be entered.

Reply:
right. that was from when I had this code in updateValue. thanks for the catch. fixed

Reply:
> I don't think newValue == oldValue will ever be true since the @Watch only fires on changes to the newValue. 

You would think, but I have seen it happen before.

Reply:
If it does, its a stencil bug :D